docs/sync-rules/NOTION.md

```diff
@@ -0,0 +1,176 @@
+
+### Example advanced sync rules
+
+#### Advanced rule for indexing every page with title containing `Demo Page`
+
+```json
+[
+  {
+    "searches": [
+      {
+        "filter": {
+          "value": "page"
+        },
+        "query": "Demo Page"
+      }
+    ]
+  }
+]
+```
+
+#### Advanced rule for indexing every database with title containing `Demo Database`
+
+```json
+[
+  {
+    "searches": [
+      {
+        "filter": {
+          "value": "database"
+        },
+        "query": "Demo Database"
+      }
+    ]
+  }
+]
+```
+
+#### Advanced rule for indexing every database with title containing `Demo Database` and every page with title containing `Demo Page`
+
+```json
+[
+  {
+    "searches": [
+      {
+        "filter": {
+          "value": "database"
+        },
+        "query": "Demo Database"
+      },
+      {
+        "filter": {
+          "value": "page"
+        },
+        "query": "Demo Page"
+      }
+    ]
+  }
+]
+```
+
+#### Advanced rule for indexing all the pages which are connected with the integration
+
+```json
+[
+  {
+    "searches": [
+      {
+        "filter": {
+          "value": "page"
+        },
+        "query": ""
+      }
+    ]
+  }
+]
+```
+
+#### Advanced rule for indexing all the pages and databases connected with the integration
+
+```json
+[
+    {
+      "searches": [
+      {
+        "query": ""
+      }
+    ]
+    }
+]
+```
+
+#### Advanced rule for indexing all the rows of the database where the record is "true" for column "Task completed" and its property(datatype) is checkbox
+
+```json
+[
+  {
+    "database_query_filters": [
+      {
+        "filter": {
+            "property": "Task completed",
+            "checkbox": {
+                "equals": true
+            }
+        },
+        "database_id": "database_id"
+      }
+    ]
+  }
+]
+```
+
+#### Advanced rule for indexing all the rows of the specific database 
+
+```json
+[
+  {
+    "database_query_filters": [
+      {
+        "database_id": "database_id"
+      }
+    ]
+  }
+]
+```
+
+#### Advanced rule for indexing all blocks defined in the "database_query_filters" and "searches"
+
+```json
+[
+    {
+    "searches" : [
+    {
+    "query": "External tasks",
+        "filter": {
+            "value": "database"
+            },
+        },
+    {
+    "query": "External tasks",
+        "filter": {
+        "value": "page"
+            }
+        }
+  ],
+
+  "database_query_filters" : [
+    {
+    "database_id": "notion_database_id1",
+    "filter": ​​{
+		"property": "Task completed",
+	      "checkbox": {
+	        "equals": true
+	      }
+	    }
+    }
+  ]
+}
+
+]
+```
+**Note** 
+For `database_query_filters`:
+    - `property` contains a string that matches the database column name.
+    - `checkbox` this is not constant depends on the column name property type it can be `date`,`files`,`formula`,`multi_select`,`number`,`people`,`phone_number`,`relation`,`rich_text`,`select`,`status`,`timestamp` or `ID`
```

Comment:
I don't think I saw this restriction captured in the JSON Schema. Why not add it?

Reply:
Since these are the list of filters already listed on Notion's documentation I have removed them now from our docs and added the link to the Notion Documentation highlighting it supports Notion filter syntax. Also, it would be quite complex for handling all of them in JSON schema when compounded with `and` `or`, therefore we can't add them all in schema.

Reply:
I was just figuring we could use a [json schema enum](https://json-schema.org/understanding-json-schema/reference/enum) for `checkbox`, no?

Reply:
[json schema enum](https://json-schema.org/understanding-json-schema/reference/enum) cannot be used here since it only works for defining allowed list of values for the keys known in advance, for example, in the below schema, enum is used to provide allowed list of values for `street_type`. However, in our case we want to define allowed list of keys for an object. 

Enum use-case:
```
{
  "type": "object",
  "properties": {
    "street_type": { "enum": ["Street", "Avenue", "Boulevard"] }
  },
}
```

Reply:
I thought that `checkbox` was always the key name, with an enum of potential values like [`date`,`files`,`formula`,`multi_select`,`number`,`people`,`phone_number`,`relation`,`rich_text`,`select`,`status`,`timestamp`, `ID`]